File: 392-is-subsequence/is-subsequence.py
class Solution:
    def isSubsequence(self, s: str, t: str) -> bool:
        ptr = 0
        for i in range(len(t)):
            if ptr == len(s):
                return True
            if s[ptr] == t[i]:
                ptr += 1
        else:
            if ptr == len(s):
                return True
            else:
                return False

File: 392-is-subsequence/is-subsequence.java
class Solution {
    public boolean isSubsequence(String s, String t) {
        int ptr = 0, n = s.length();
        for (int i = 0; i < t.length(); i++) {
            if (ptr < n && s.charAt(ptr) == t.charAt(i)) {
                ptr += 1;
            } else if (ptr == n) {
                return true;
            }

        }
        if (ptr == n) {
            return true;
        } else {
            return false;
        }
    }
}